### ISSUE TITLE
Explore page - Searches don't terminate, need to be Cancelled

### ISSUE THREAD

**Issue by Andrew-Clement** [ixmaps/website#25]
Two searches that should be legit, or at least produce an an alert, don't complete and have to be cancelled.  In the first case, search on submitter "andrewC" (I missed selecting "AndrewC" from the autocomplete, but even if search needs to be case specific, A message of Not Found would be appropriate. From the console output (below) it appears the parsing doesn't have sufficient error handling.

![pastedgraphic-1](https://cloud.githubusercontent.com/assets/12848029/20034763/54c89b98-a387-11e6-8b1e-81ea6f0ea5bf.png)

The second case produces the same results, but may have a different cause.  The postal code I mentioned has been used for contributing many routes ( see other recent screen shots in Client issues), but did not appear as possible selection in the autocomplete,.  Evidently it isn't being recorded properly.

![pastedgraphic-2](https://cloud.githubusercontent.com/assets/12848029/20034762/54c537dc-a387-11e6-9d30-8713f057f47a.png)
![pastedgraphic-3](https://cloud.githubusercontent.com/assets/12848029/20034764/54ca5a46-a387-11e6-8d7f-08334ece934f.png)

**Comment by Andrew-Clement** [comment]
I'm encountering this same problem, of a search not responding and having to be Canceled, in a number of other situations.
eg. search on country that appears as a type ahead option (in this case AM) (See also #27)  
﻿
![pastedgraphic-3](https://cloud.githubusercontent.com/assets/12848029/20466303/82ef8a46-af25-11e6-86d7-8047b9ec78fb.png)

**Comment by Andrew-Clement** [comment]
Suggest that as an interim measure until the underlying issues can be sorted out, provide a timeout feature,

**Comment by Andrew-Clement** [comment]
And another example of non completion, in this case there should be no results, due to an inadvertent mix-formulation of the search query
![pastedgraphic-4](https://cloud.githubusercontent.com/assets/12848029/20469915/b0596d34-af58-11e6-86fb-199f5671c474.png)

**Comment by agamba** [comment]
Hi Andrew, you are right this is a bug in the search. I believe it's happening when no results are found. Will inspect and report asap.

**Comment by Andrew-Clement** [comment]
Good. Thanks Antonio.

The priority of course is to gracefully handle the situation when there are no results to be found.  

Then there are the cases where there are results that should be found, but weren't.

**Comment by agamba** [comment]
This has been fixed. I've corrected a bug that was returning invalid data and causing the search didn't completed.

I've tested the case reported before and now it completes returning no data, as it should. Note that if your have in the rules two originating cities, no TR will match this rule. But the good thing is that now it does not crash.

does : originate : city : Toronto : AND | Traceroutes: 14874
does : terminate : city : Toronto : AND | Traceroutes: 9411
does : goVia : city : Chicago : AND | Traceroutes: 44318
does : originate : city : Chicago : AND | Traceroutes: 1
Total traceroutes : 0

**Comment by Andrew-Clement** [comment]
Good to see this Anto!


On 2017-02-08, at 2:12 PM, Antonio Gamba-Bari <notifications@github.com> wrote:

This has been fixed. I've corrected a bug that was returning invalid data and causing the search didn't completed.

I've tested the case reported before and now it completes returning no data, as it should. Note that if your have in the rules two originating cities, no TR will match this rule. But the good thing is that now it does not crash.

does : originate : city : Toronto : AND | Traceroutes: 14874
does : terminate : city : Toronto : AND | Traceroutes: 9411
does : goVia : city : Chicago : AND | Traceroutes: 44318
does : originate : city : Chicago : AND | Traceroutes: 1
Total traceroutes : 0
